src/components/StudioApp.tsx

```
@@ -1761,69 +1779,22 @@ export default function StudioApp() {
-    ]
-  );
-
-  const tabLabel = NAV.find((n) => n.key === tab)?.label ?? "";
-  const tabSub = NAV.find((n) => n.key === tab)?.sub ?? "";
+  // Split commits itself: while the Split stage is open, the current cut set
+  // is the committed one. Re-running when the signature changes keeps Match
+  // and Join in step without a separate "Commit" click.
+  const shouldAutoCommitSplit = tab === "split" && splitSegments.length > 0 && !isCommittedSplitCurrent;
+  useEffect(() => {
+    if (!shouldAutoCommitSplit) return;
+    handleCommitSplit();
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/components/StudioApp.tsx:1788:5
  1786 |   useEffect(() => {
  1787 |     if (!shouldAutoCommitSplit) return;
> 1788 |     handleCommitSplit();
       |     ^^^^^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  1789 |     // handleCommitSplit closes over the latest split state; the signature guard prevents re-runs.
  1790 |     // eslint-disable-next-line react-hooks/exhaustive-deps
  1791 |   }, [shouldAutoCommitSplit, splitSignature]);

(react-hooks/set-state-in-effect)